lib/gooddata/models/metadata/report.rb

```diff
@@ -195,5 +208,33 @@ def replace(what, for_what)
       new_defs.pmap(&:save)
       self
     end
+
+    ## Update report definition and reflect the change in report
+    #
+    # @param [Hash] opts Options
+    # @option opts [Boolean] :immutable (true) Immutable true means that new definition will be created
+    # @return [GoodData::ReportDefinition] Updated and saved report definition
+    def update_definition(opts = { :immutable => true }, &block)
```

Comment:
Could we change this :immutable param to something like :add_report_definition ? I think it could represent better what it does (if I understand the code correctly).